aws-synthetics-canary/src/main/java/com/amazon/synthetics/canary/CreateHandler.java

```diff
@@ -159,7 +168,7 @@
         try {
             proxy.injectCredentialsAndInvokeV2(createCanaryRequest, syntheticsClient::createCanary);
         } catch (final ValidationException e) {
-            throw new CfnInvalidRequestException(e.getMessage());
+            throw new CfnAlreadyExistsException(ResourceModel.TYPE_NAME, model.getPrimaryIdentifier().toString());
```

Comment:
How do you know that a validation exception is always because it already exists? Shouldn't we check to make sure the `ValidationException` is *actually* a duplicate error, not just a regular validation problem?

Reply:
There is a broader issue here as we do not have the granularity of exceptions in Synthetics. While we are fixing the service side exceptions, our contract tests are not passing due to this particular test.

In this case before we hit this validation, Schema validation would have validated all before submitting the request and would have failed and that is reason to proceed further for now until we fix the API 

This will eventually changed to only catch CanaryAlreadyExistsException soon.